src/main/java/io/cdap/plugin/http/source/common/BaseHttpSourceConfig.java

```diff
@@ -941,6 +960,13 @@ PAGINATION_INDEX_PLACEHOLDER, getPaginationType()),
     }
   }
 
+  private boolean refreshTokenGrantType() {
+    if (getGrantType() == "refresh_token") {
```

Comment:
create an enum for grant types and change the above code also accordingly. and write grant type specific if conditions. e.g.(if grant == enum.refreshtoken) else if grant=client_credentials

Reply:
Enum for grant type created and made changes to the conditions accordingly.